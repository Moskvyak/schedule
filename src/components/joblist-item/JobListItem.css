.joblist-item__wrapper {
  border-left: 5px solid green;
  line-height: 2em;
  border-bottom: 1px solid #ccc;
  margin-bottom: 10px;
  padding-left: 20px;
}

.joblist-item__time-period {
  font-size: 1em;
  color: #757575;
}
.joblist-item__title {
  font-size: 1.2em;
  color: #424242;
}
.joblist-item__title span {
  color: #212121;
}

