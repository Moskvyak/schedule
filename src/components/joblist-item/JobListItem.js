import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

import { formatLocalDateTime } from '../../utils/formatDate';
import './JobListItem.css';

export const JobListItem = React.memo(({ option }) => {
  const { name, start, end, contact } = option;
  const startDate = formatLocalDateTime(moment(start));
  const endDate = formatLocalDateTime(moment(end));
  return (
    <div className="joblist-item__wrapper">
      <div className="joblist-item__title">
        <span>{contact.name}</span> - {name}
      </div>
      <div className="joblist-item__time-period">
        {startDate} - {endDate}
      </div>
    </div>
  );
});

JobListItem.propTypes = {
  option: PropTypes.shape({
    name: PropTypes.string.isRequired,
    start: PropTypes.string.isRequired,
    end: PropTypes.string.isRequired,
    contact: PropTypes.shape({
      name: PropTypes.string.isRequired,
    })
  }).isRequired
}