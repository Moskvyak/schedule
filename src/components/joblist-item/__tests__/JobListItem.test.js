import React from 'react';
import renderer from 'react-test-renderer';
import { JobListItem } from '../JobListItem';

describe('<JobListItem />', () => {
  const requiredProps = {
    option: {
      name: 'test',
      start: '2018-09-01T11:00:00Z',
      end: '2018-09-01T12:00:00Z',
      contact: { name: 'testName' }
    }
  };

  it('should render with required props', () => {
    const component = renderer.create(<JobListItem {...requiredProps} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
