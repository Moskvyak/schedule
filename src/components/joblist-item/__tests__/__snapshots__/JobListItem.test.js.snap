// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JobListItem /> should render with required props 1`] = `
<div
  className="joblist-item__wrapper"
>
  <div
    className="joblist-item__title"
  >
    <span>
      testName
    </span>
     - 
    test
  </div>
  <div
    className="joblist-item__time-period"
  >
    09/01/2018 11:00 AM
     - 
    09/01/2018 12:00 PM
  </div>
</div>
`;
